synapse/types/__init__.py

```diff
@@ -69,6 +69,8 @@
 MutableStateMap = MutableMapping[StateKey, T]
 
 # JSON types. These could be made stronger, but will do for now.
+# A "simple" (canonical) JSON value.
+SimpleJsonValue = Optional[Union[str, int, bool]]
```

Comment:
FTR since I checked: JSON supports float through `Number` type, but [Canonical JSON](https://spec.matrix.org/v1.5/appendices/#canonical-json) explicitly doesn't.

Reply:
The MSC calls this out, FWIW.